client/app/visualizations/chart/Editor/ColumnMappingSelect.jsx

```diff
@@ -0,0 +1,60 @@
+import { isString, map, uniq, flatten, filter, sortBy, keys } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import Select from 'antd/lib/select';
+
+const MappingTypes = {
+  x: { label: 'X Column' },
+  y: { label: 'Y Columns', multiple: true },
+  series: { label: 'Group by' },
+  yError: { label: 'Errors column' },
+  size: { label: 'Bubble size column' },
+  zVal: { label: 'Color Column' },
+};
+
+export default function ColumnMappingSelect({ value, availableColumns, type, onChange }) {
+  const options = sortBy(filter(
+    uniq(flatten([availableColumns, value])),
+    v => isString(v) && (v !== ''),
+  ));
+  const { label, multiple } = MappingTypes[type];
+
+  return (
+    <div className="m-b-15">
+      <label htmlFor={`chart-editor-column-mapping-${type}`}>{label}</label>
+      <Select
+        id={`chart-editor-column-mapping-${type}`}
+        className="w-100"
+        data-test={`Chart.ColumnMapping.${type}`}
+        mode={multiple ? 'multiple' : 'default'}
+        allowClear
+        placeholder={multiple ? 'Choose columns...' : 'Choose column...'}
+        value={value}
+        onChange={column => onChange(column, type)}
+      >
+        {map(options, c => (
+          <Select.Option key={c} value={c} data-test={`Chart.ColumnMapping.${type}.${c}`}>{c}</Select.Option>
+        ))}
```

Comment:
Something is causing the "allowClear" icon to appear on hover, even when there's no option selected.

Reply:
Strange. I'll check

Reply:
Fixed